src/hiro_graph_client/batchclient.py

```diff
@@ -843,62 +838,44 @@ def __init__(self,
         Use the connection to HIRO HiroGraph either by giving a predefined *hiro_token* or by
         specifying all other parameters needed for authentication.
 
-        :param graph_endpoint: required: URL of the graph.
+        :param endpoint: required: URL of the graph.
+        :param token_handler: Supply a dedicated token_handler.
         :param callback: required when multi_command() is used, optional otherwise: Callback object for results.
-        :param hiro_token: optional, required if other authentication data is missing: Predefined token to authenticate
-                           against HIRO. Overrides all other authentication data.
-        :param username: optional, required if *hiro_token* is None: Username for HiroGraph .
-        :param password: optional, required if *hiro_token* is None: Password for HiroGraph.
-        :param client_id: optional, required if *hiro_token* is None: Id for Authentication (OAuth2).
-        :param client_secret: optional, required if *hiro_token* is None: Secret for Authentication (OAuth2).
-        :param auth_endpoint: optional, required if *hiro_token* is None: URL of the authentication API.
         :param use_xid_cache: Use xid caching. Default is True when omitted or set to None.
         :param proxies: Proxy configuration for *requests*. Default is None.
         :param parallel_workers: Amount of parallel workers for requests. Default is 8.
         :param queue_depth: Amount of entries the *self.request_queue* and *self.result_queue* can hold. Default is to
                             set it to the same value as *parallel_workers*.
         """
+        super().__init__(endpoint, True, proxies)
+
+        if not endpoint:
+            raise ValueError('Required attribute "endpoint" for HIRO Graph API is not set.')
```

Comment:
exception should be raised in super class since the field's value will be assigned there